src/server.c

```diff
@@ -680,6 +680,52 @@ int clientsCronResizeQueryBuffer(client *c) {
     return 0;
 }
 
+/* The client output buffer can be adjusted to better fit the memory requierments.
+ *
+ * the logic is:
+ * in case the last observed peak size of the buffer equals the buffer size - we double the size
+ * in case the last observed peak size of the buffer is less than half the buffer size - we shrink by half.
+ * The buffer peak will be reset back to zero every 5 seconds
+ * The function always returns 0 as it never terminates the client. */
+int clientsCronResizeOutputBuffer(client *c) {
+
+    size_t new_buffer_size = 0;
+    const size_t buffer_target_shrink_size = c->buf_usable_size/2;
+    const size_t buffer_target_expend_size = c->buf_usable_size*2;
```

Comment:
for growing, it makes sense to grow incrementally, i.e. each time the peak==usable it means we reached the max we could reach, and we can grow to *2.
but for shrinking, if a client suddenly became silent, or started issuing small commands, there's no need to incrementally shrink it by half each time.
maybe we can do better, like shrink it to the size of the peak? or maybe to double the size of the recent peak?

Reply:
@oranagra I agree. that sound reasonable. I can change it to shrink to last peak size + 1

Reply:
for a moment i preferred last peak * 2, but actually, malloc is likely to give you some 30% extra, so +1 is good.

Reply:
Great - @oranagra can you please help me in running the tests again?
I cannot trigger them myself and I think the results are outdated